src/renderer/classes/outputManager/MemberChannel.js

```diff
@@ -0,0 +1,106 @@
+import '../../typedef';
+import { NOTE_ON, MODULATION, NOTE_OFF } from './note-actions';
+import { deriveStatusByte } from '../../utils/midi/statusByteUtils';
+import * as midiMessageType from '../../utils/midi/midi-message-types';
+import * as defaults from '../../constants/defaults';
+
+class MemberChannel {
+  /**
+   * @param  {number} midiChannel integer from 1 to 16
+   * @param  {function} nowFn function to current time. e.g. performance.now
+   */
+  constructor (midiChannel, nowFn) {
+    this.nowFn = nowFn;
+    this.timeOfLastNoteOff = nowFn();
+    this.activeNoteOn = null;
+    this.midiChannel = midiChannel;
+  };
+  /**
+   * @param {string} actionType
+   * @param  {NoteOn | Modulation | NoteOff } [action]
+   */
+  deriveMidiMessage (actionType, action) {
+    this.messages = [];
+    switch (actionType) {
+      case NOTE_ON:
+        const noteOn = Object.assign({}, defaults.NOTE_ON, action);
+        if (this.isOccupied) {
+          return [this.deriveNoteOffMessage(), ...this.deriveNoteOnRelatedMessages(noteOn)];
+        }
+        return this.deriveNoteOnRelatedMessages(noteOn);
+      case MODULATION:
+        if (!this.isOccupied) return [];
+        // TODO
+        return [];
+      case NOTE_OFF:
+        if (!this.isOccupied) return [];
+        // TODO
+        return [];
+    }
+  };
+
+  isOccupied () {
+    return this.activeNoteOn != null;
```

Comment:
Use
```javascript
return Boolean(this.activeNoteOn);
```
or
```javascript
return !!this.activeNoteOn
```

for the same reason of redundant negations.

Reply:
Ok, I will use the first one.